src/Sylius/Bundle/ApiBundle/Applicator/ProductReviewStateMachineTransitionApplicator.php

```diff
@@ -53,7 +54,13 @@ public function reject(ReviewInterface $data): ReviewInterface
 
     private function applyTransition(ReviewInterface $review, string $transition): void
     {
-        $this->getStateMachine()->apply($review, ProductReviewTransitions::GRAPH, $transition);
+        $stateMachine = $this->getStateMachine();
+
+        if (false === $stateMachine->can($review, ProductReviewTransitions::GRAPH, $transition)) {
```

Comment:
Why is it better this way instead of just using a simple try-catch block?

Reply:
Because the `try-catch` block catches all exceptions from the Symfony workflow, and the method `can` only checks if a transition is possible. In the `try-catch` block, we always receive the same exception, whereas in the second case, we receive different exceptions if the method `apply` throws an exception.